alerting/src/main/kotlin/org/opensearch/alerting/AlertService.kt

```diff
@@ -758,6 +845,37 @@ class AlertService(
         return searchResponse
     }
 
+    /**
+     * Searches for Alerts in the monitor's alertIndex.
+     *
+     * @param monitorId The Monitor to get Alerts for
+     * @param size The number of search hits (Alerts) to return
+     */
+    private suspend fun searchAlerts(
+        workflow: Workflow,
+        size: Int,
+        dataSources: DataSources,
+    ): SearchResponse {
+        val workflowId = workflow.id
+        val alertIndex = dataSources.alertsIndex
+
+        val queryBuilder = QueryBuilders.boolQuery()
+            .must(QueryBuilders.termQuery(Alert.WORKFLOW_ID_FIELD, workflowId))
+            .must(QueryBuilders.termQuery(Alert.MONITOR_ID_FIELD, ""))
```

Comment:
can there be a doc which has both `workflowId` & `monitorId` field set?

Reply:
Alert created from composite monitor trigger will have only empty `monitorId`.

The underlying alerts created in AUDIT state will have both `monitorId` and `worklfowId`.